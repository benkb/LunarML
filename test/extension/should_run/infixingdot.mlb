$(SML_LIB)/basis/basis.mlb
ann "allowInfixingDot true" in
infixingdot.sml
end

