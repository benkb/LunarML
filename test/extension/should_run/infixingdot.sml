infix 3 .foo.
infixr 4 .bar.
fun a .foo. b = "foo(" ^ a ^ "," ^ b ^ ")"
fun a .bar. b = "bar(" ^ a ^ "," ^ b ^ ")";
print (("a" .foo. "b" .foo. "c" .foo. "d") ^ "\n");
print (("a" .foo. "b" .foo. "c" .bar. "d") ^ "\n");
print (("a" .foo. "b" .bar. "c" .foo. "d") ^ "\n");
print (("a" .foo. "b" .bar. "c" .bar. "d") ^ "\n");
print (("a" .bar. "b" .foo. "c" .foo. "d") ^ "\n");
print (("a" .bar. "b" .foo. "c" .bar. "d") ^ "\n");
print (("a" .bar. "b" .bar. "c" .foo. "d") ^ "\n");
print (("a" .bar. "b" .bar. "c" .bar. "d") ^ "\n");
structure S = struct
val foo = foo
val bar = bar
end;
print (("a" .S.foo. "b" .S.foo. "c" .S.foo. "d") ^ "\n");
print (("a" .S.foo. "b" .S.foo. "c" .S.bar. "d") ^ "\n");
print (("a" .S.foo. "b" .S.bar. "c" .S.foo. "d") ^ "\n");
print (("a" .S.foo. "b" .S.bar. "c" .S.bar. "d") ^ "\n");
print (("a" .S.bar. "b" .S.foo. "c" .S.foo. "d") ^ "\n");
print (("a" .S.bar. "b" .S.foo. "c" .S.bar. "d") ^ "\n");
print (("a" .S.bar. "b" .S.bar. "c" .S.foo. "d") ^ "\n");
print (("a" .S.bar. "b" .S.bar. "c" .S.bar. "d") ^ "\n");
