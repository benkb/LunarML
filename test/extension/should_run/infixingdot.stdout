foo(foo(foo(a,b),c),d)
foo(foo(a,b),bar(c,d))
foo(foo(a,bar(b,c)),d)
foo(a,bar(b,bar(c,d)))
foo(foo(bar(a,b),c),d)
foo(bar(a,b),bar(c,d))
foo(bar(a,bar(b,c)),d)
bar(a,bar(b,bar(c,d)))
foo(foo(foo(a,b),c),d)
foo(foo(a,b),bar(c,d))
foo(foo(a,bar(b,c)),d)
foo(a,bar(b,bar(c,d)))
foo(foo(bar(a,b),c),d)
foo(bar(a,b),bar(c,d))
foo(bar(a,bar(b,c)),d)
bar(a,bar(b,bar(c,d)))
